src/Pack.js

```diff
@@ -0,0 +1,194 @@
+*/
+export default class Pack extends Viz {
+
+  /**
+      @memberof Pack
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._layoutPadding = 1;
+    this._on.mouseenter = () => {};
+    this._on["mouseleave.shape"] = () => {
+      this._tooltip = false;
+      this.hover(false);
+    };
+
+    let hoverData = [];
+    const recursionCircles = d => {
+      if (d.values) {
+        d.values.forEach(h => {
+          hoverData.push(h);
+          recursionCircles(h);
+        });
+      }
+      else {
+        hoverData.push(d);
+      }
+    };
+    const defaultMouseMove = this._on["mousemove.shape"];
+    this._on["mousemove.legend"] = (d, i) => {
+      const ids = this._ids(d, i);
+      hoverData = [d];
+      recursionCircles(d);
+
+      this.hover(h => {
+        const _hover = Object.keys(h).filter(key => key !== "value").every(key => d[key] && d[key].includes(h[key]));
+
+        if (_hover) hoverData.push(h);
+        else if (ids.includes(h.key)) {
+          hoverData.push(h);
+          recursionCircles(h);
+        }
+
+        return hoverData.includes(h);
+      });
+
+    };
+    this._on["mousemove.shape"] = (d, i) => {
+      defaultMouseMove(d, i);
+
+      this._tooltip = d.__d3plusTooltip__;
+
+      hoverData = [d];
+      recursionCircles(d);
+
+      this.hover(h => hoverData.includes(h));
+
+    };
+    this._pack = pack();
+    this._packOpacity = 0.25;
+    this._shape = constant("Circle");
+    this._shapeConfig = assign(this._shapeConfig, {
+      Circle: {
+        opacity: d => d.__d3plusOpacity__ || 1,
+        labelConfig: {
+          fontResize: true
+        }
+      }
+    });
+    this._sort = (a, b) => b.value - a.value;
+    this._sum = accessor("value");
+    this._tooltip = false;
+
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const diameter = Math.min(height, width);
+    const transform = `translate(${(width - diameter) / 2}, ${(height - diameter) / 2})`;
+
+    let nestedData = nest();
+    for (let i = 0; i <= this._drawDepth; i++) nestedData.key(this._groupBy[i]);
+    nestedData = nestedData.entries(this._filteredData);
+
+    const packData = this._pack
+      .padding(this._layoutPadding)
+      .size([diameter, diameter])
+      (hierarchy({key: nestedData.key, values: nestedData}, d => d.values).sum(this._sum).sort(this._sort))
+      .descendants();
+
+    packData.forEach((d, i) => {
+      d.__d3plus__ = true;
+      d.children = d.children;
+      d.data.__d3plusTooltip__ = !d.height ? true : false;
```

Comment:
move this line to come after the `__d3plusOpacity__` line and format them the same way (pick either a single if statement or this ternary format)